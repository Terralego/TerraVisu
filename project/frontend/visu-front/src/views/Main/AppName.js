import React from 'react';

export const AppName = ({ title = 'TerraVisu', version = 'v1.0' }) => (
  <div className="bp3-dark appName">
    <span className="appName-title">
      { title }
    </span>
    <span className="appName-version">{ version }</span>
  </div>
);


export default AppName;
