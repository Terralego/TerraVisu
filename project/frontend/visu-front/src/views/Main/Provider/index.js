import MainProvider from './MainProvider';

export default MainProvider;
