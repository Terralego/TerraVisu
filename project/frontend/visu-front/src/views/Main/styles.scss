@import '../../theme.scss';
@import '../variables.scss';

$navWidth: 4rem;
$layersTreeWidth: 22.5rem;
$widgetsWidth: 20rem;
$detailsWidth: 25.5rem;
$mapboxContributionsWidth: 18.5rem;
$mapboxControlScaleWidth: 4rem;

$tableHeight: 33vh;


.main {
  display: flex;
}

.main .map-navigation {
  padding: 0 ! important;

  &__header {
    .appName-title-logo {
      margin-top: 0;
    }
  }
}

.layerstree-group:first-child {
  margin-top: 0;
}

.layerstree-panel-list {
  width: 100%;
}

.main .layerNode-options__buttons--active, .layerNode-widgets__buttons--active {
  .icon, svg {
    filter: invert(0.5) sepia(1) saturate(5) hue-rotate(140deg);
  }
}

.details {
  .details__info-administrative {
    text-align: center;
    margin-bottom: 4px;
  }

  .details__list {

    .details__column-label {
      white-space: normal;
      font-size: .85rem;
      width: 15.5rem;
      padding-right: .5rem;
    }

    .details__column-value {
      vertical-align: bottom;
    }
  }
}

// Fix temporary waiting upgrade TerraLego
.visualizer {
  .interactive-map__legends {

    .tf-legend {
      max-width: none;
    }

    .tf-legend__title {
      align-self: center;
    }
  }
}

// Override TerraLego
.mapboxgl-ctrl:hover {
  color: #202B33;
}

.interactive-map {
  .tf-legend {
    margin-bottom: 1em;
  }
}

.visualizer {
  .mapboxgl-control-container {
    transition: all .2s ease-in;
  }

  .mapboxgl-ctrl-top-left,
  .mapboxgl-ctrl-bottom-left {
    transform: none;
    transition: all .2s ease-in;
  }

  .mapboxgl-ctrl-bottom-left {
    transform: none;
    margin: -0.5rem 0.5rem;
    left: 0rem;
    right: 22rem;
    transition: all .2s ease-in;

    .mapboxgl-ctrl {
      float: none;
    }
  }

  &.visualizer--with-layers-tree {
    .mapboxgl-ctrl-bottom-left {
      transform: none;
      margin: -0.5rem 0.5rem;
      left: $layersTreeWidth;
      right: 22rem;
      transition: all .2s ease-in;
    }

    &.visualizer--with-table {
      .mapboxgl-ctrl-bottom-left {
        left: 0;
        transition: none;
      }

      .mapboxgl-ctrl-logo {
        right: 19.5rem;
        transition: none;
      }
    }
  }

  .mapboxgl-popup {

    .mapboxgl-popup-content {
      text-align: center;
    }
  }
}

// Override Mapbox position logo
.visualizer {

  .mapboxgl-ctrl-logo {
    position: absolute;
    right: 10px;
    bottom: 15px;
    transition: all .2s ease-in;

  }

  &.visualizer--with-layers-tree {
    .mapboxgl-ctrl-logo {
      right: -38px;
      bottom: 15px;
    }
  }

}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {
  .visualizer {
    .interactive-map__legends {
      right: 3.25rem;
    }
  }
}

@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .visualizer {
    .interactive-map__legends {
      right: 3.75rem;
    }
  }
}
