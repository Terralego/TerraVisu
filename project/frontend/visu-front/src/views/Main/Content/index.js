import React, { Suspense, lazy } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import compose from '@terralego/core/utils/compose';

import withEnv from '../../../config/withEnv';
import { connectSettings } from '../Provider/context';
import Loading from '../../../components/Loading';
import AppName from '../AppName';
import VisualizerLoading from '../../Visualizer/Loading';

const Profile = lazy(() => import('../../Profile'));
const Error404 = lazy(() => import('../../Error404'));
const Visualizer = lazy(() => import('../../Visualizer'));


const Content = ({
  env: { VIEW_ROOT_PATH = 'view', DEFAULT_VIEWNAME = 'rechercher' },
  settings: { title, version, credits = 'credits', theme: { logoUrl } = {} },
}) => {
  const exportCallback = React.useCallback((xlsx, sheet) => {
    xlsx.utils.sheet_add_aoa(sheet, [
      [],
      [credits],
    ], { origin: -1 });
    const wholeRange = xlsx.utils.decode_range(sheet['!ref']);
    const lastCell = sheet[xlsx.utils.encode_cell({
      c: 0,
      r: wholeRange.e.r,
    })];
    lastCell.l = {
      Target: logoUrl,
      Tooltip: title,
    };
  }, [credits, logoUrl, title]);

  return (
    <div className="main__content">
      <Switch>
        <Route path="/create-account/:id/:token">
          <Suspense fallback={<Loading />}>
            <Profile />
          </Suspense>
        </Route>
        <Route path={`/${VIEW_ROOT_PATH}/not-found`}>
          <p>Vue non trouv√©e :(</p>
        </Route>
        <Route path={`/${VIEW_ROOT_PATH}/:viewName`}>
          <Suspense fallback={<VisualizerLoading />}>
            <Visualizer
              renderHeader={<AppName title={title} version={version} />}
              exportCallback={exportCallback}
            />
          </Suspense>
        </Route>
        <Route path="/">
          <Redirect to={`/${VIEW_ROOT_PATH}/${DEFAULT_VIEWNAME}`} />
        </Route>
        <Route render={() => (
          <Suspense fallback={<VisualizerLoading />}>
            <Error404 />
          </Suspense>
        )}
        />
      </Switch>
    </div>
  );
};

export default compose(
  withEnv,
  connectSettings('settings'),
)(Content);
