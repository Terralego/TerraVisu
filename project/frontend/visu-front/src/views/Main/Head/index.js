import { connectSettings } from '../Provider/context';
import Head from './Head';


export default connectSettings('settings')(Head);
