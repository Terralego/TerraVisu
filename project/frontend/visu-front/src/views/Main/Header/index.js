import { connectMain, connectSettings } from '../Provider/context';
import withEnv from '../../../config/withEnv';

import Header from './Header';

export default withEnv(connectMain('isHeaderOpen', 'toggleHeader')(connectSettings('settings')(Header)));
