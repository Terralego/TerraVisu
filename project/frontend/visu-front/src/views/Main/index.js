import React from 'react';

import Header from './Header';
import Content from './Content';
import MainProvider from './Provider/MainProvider';
import Head from './Head';
import './styles.scss';
import SettingsProvider from './Provider/SettingsProvider';

export const Main = () => (
  <SettingsProvider>
    <MainProvider>
      <Head />
      <main className="main">
        <Header />
        <Content />
      </main>
    </MainProvider>
  </SettingsProvider>
);
export default Main;
