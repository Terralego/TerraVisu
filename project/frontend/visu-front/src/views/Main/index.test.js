import React from 'react';
import renderer from 'react-test-renderer';
import Main from '.';

jest.mock('./Header', () => () => <div>Header</div>);
jest.mock('./Content', () => () => <div>Content</div>);

it('should render correctly', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
