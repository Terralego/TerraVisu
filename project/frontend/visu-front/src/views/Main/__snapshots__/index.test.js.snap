// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<main
  className="main"
>
  <div>
    Header
  </div>
  <div>
    Content
  </div>
</main>
`;
